README.md

```diff
@@ -1,36 +1,54 @@
-# README
-
-```shell
-npm i
-```
-
-Enable the Google Apps Script API at https://script.google.com/home/usersettings:
+# SD Onestop to Basecamp Integration
+Takes events from the Onestop's daily tabs and populates Basecamp with appropriate todos for each person.
 
+## Developer setup
+### Local development environment
+1.  ```npm i ```
+2. Enable the Google Apps Script API at https://script.google.com/home/usersettings:
 ![Enable Apps Script API](https://user-images.githubusercontent.com/744973/54870967-a9135780-4d6a-11e9-991c-9f57a508bdf0.gif)
-
-```shell
-clasp login
+3. Download clasp: ```npm install -g @google/clasp```
+4.  ```clasp login ```
+
+### Personal Onestop and app script setup
+5. Make your own copy of the Onestop from [this folder](https://drive.google.com/drive/folders/1s_u-hmstlLL1JqRKyKJ7r2AMyyTCCiut) and add your name to it.
+6. In your new sheet, on the menu click Extensions > App Script
+7. Select "Create a new project"
+8. Name it appropriately, e.g. "Onestop Basecamp Integration - My Name"
+9. In the Apps Scripts project, open "Project Settings" on the left bar
+10. Copy your Script ID.
+11. Replace `scriptId` in [`.clasp.json`](./.clasp.json) with your created script ID.
+
+### Personal Basecamp integration setup
+12. Login to [Basecamp](https://launchpad.37signals.com/signin) with your A2N Google account. Ensure you have access to the Acts2Network project before continuing.
+13. Open the [Basecamp Integrations page](https://launchpad.37signals.com/integrations)
+14. Click "Register an application"
+15. Fill out application details:
+- Name it appropriately (e.g. Onestop Integration - My Name)
+- Company and website URL don't matter
+- Select Basecamp 4 under Products
+- For OAuth2 redirect URI, fill in the URI with the following, and replace `YOUR_SCRIPT_ID` with your Google Apps Scripts ID from the earlier section: `https://script.google.com/macros/d/YOUR_SCRIPT_ID/usercallback`. 
+16. Click "Register this app"
+17. Open your app and copy your Client ID and Client Secret for the following step.
+19. Create a new file `environmentVariables.ts` inside the [config](./config/) folder, with the following contents:
+```js
+export const BASECAMP_CLIENT_ID = 'REPLACE_WITH_YOUR_BASECAMP_CLIENT_ID';
+export const BASECAMP_CLIENT_SECRET = 'REPLACE_WITH_YOUR_BASECAMP_CLIENT_SECRET';
```

Comment:
Not a blocking review comment, just a question. Is this just temporary or is the plan to hardcode these values?

Reply:
i haven't fully thought it through yet, we'll have a more solid approach in a future task. but here's the factors i'm considering:
- we should avoid putting secrets in version control
- for us to all develop independently, each of ours will need to be separate, hence this is local for each of us and in gitignore
- but we need to figure out how to do this for "prod", e.g. the real onestop